packages/core/src/styled.ts

```diff
@@ -23,5 +21,127 @@ function styled<T extends keyof JSX.IntrinsicElements>(Component: T) {
   };
   return fn;
 }
+/**
+ * Kuma UI's `styled` API allows you to create styled React components using tagged template literals. This makes it a familiar and comfortable choice for developers who have worked with libraries like styled-components or Emotion.
+ */
+const styled = Object.assign(_styled, {
```

Comment:
While I understand the intent behind this implementation, I have some concerns regarding its maintainability, especially when new tags are introduced or if certain tags become deprecated. 
I'll give it some thought and see if we can come up with a more elegant solution tomorrow. (maybe utilizing `Proxy`?)

Reply:
Yeah, Proxy would work. Why don't I push a commit with Proxy to see how it looks.